src/ngraph_builder1.h

```diff
@@ -0,0 +1,138 @@
+
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/graph/algorithm.h"
+
+using namespace std;
+namespace ng = ngraph;
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+// TODO (sarkars): make sure all comments from old builder are copied correctly.
+
+// The purpose of Builder is to accept a TF graph and convert it to a
+// equivalent ngraph fucntion. To that end, it acts as an interface to
+// a library of 'TranslateOps' functions. These functions are defined
+// in ngraph_translateops.h. Builder just goes through each TF op and
+// calls the appropriate 'TranslatorFn'. It does not implement the per-op
+// translation, the heavy-lifting being done by ngraph_translateops.h;
+// It merely does some bookkeeping, such as process the nodes in correct
+// order, keep track of and retrieve parent nGraph nodes etc.
+
+class Builder1 {
+  using VectNg = std::vector<shared_ptr<ng::Node>>;
+  using TranslatorFn = function<Status(
+      const Node*, VectNg&, const std::vector<const Tensor*>&, VectNg&)>;
+
+ public:
+  // Note: in case we want to get rid of Initialize, pass an
+  // OpKernelConstruction* ctx to the constructor, and use OP_REQUIRES_OK
+  // instead of TF_RETURN_IF_ERROR. This gets rid of Initialize() as it
+  // can handle errors in construction.
+  // But in the case of overloaded constructor that does not accept a ctx,
+  // which is used for OpExecuter test class, we cannot handle error during
+  // construction.
+  // Hence keeping the Initialize() function
+
+  // This constructor is for actual use (encapsulate_op)
+  Builder1(const Graph& tf_graph, OpKernelConstruction* ctx)
+      : tf_graph(tf_graph) {}
+
+  // And this version is for tests (OpExecuter)
+  Builder1(const Graph& tf_graph) : Builder1(tf_graph, nullptr) {}
+
+  // TranslateGraph is overloaded. This is for actual use (encapsulate_op)
+  Status TranslateGraph(OpKernelContext* ctx,
+                        std::shared_ptr<ngraph::Function>& ng_function);
+
+  // And this version is for tests (OpExecuter)
+  Status TranslateGraph(const std::vector<TensorShape>&,
+                        const std::vector<const Tensor*>&,
+                        shared_ptr<ng::Function>&);
+
+ private:
+  // The op map holds a mapping from TensorFlow op names (strings) to
+  // vector of generated nGraph nodes. Since we process nodes in toposort
+  // order, it is guaranteed when processing a TF node, its parents
+  // will already have been processed and safely stowed in ng_op_map
+  std::unordered_map<std::string, VectNg> ng_op_map;
+
+  // Prevent Initialize from running twice
+  bool is_initialized = false;
+  const Graph& tf_graph;
+
+  // An array that will be useful in creating the static_input_map
+  std::vector<bool> m_input_is_static;
+
+  // Vectors containing TF nodes in 3 bins: inputs, outputs, actual ops
+  vector<const Node *> tf_params, tf_ret_vals, tf_ops;
```

Comment:
As you refactor this feel free to revisit the question of whether and where the `tf_` and `ng_` prefixes are necessary. I think they're really only needed where there's a need to distinguish between TensorFlow's notion of a `Foo` and nGraph's notion of a `Foo` (e.g., TF functions and nGraph functions).

Reply:
The 'GetInputParams' and 'GetOutputNodes' works with ng and tf nodes, so I think the tf_ or ng_ prefix is useful here to distinguish between ng in/outs and tf in/outs.